proj_test.py

```diff
@@ -0,0 +1,169 @@
+import unittest
+import enc_dec
+
+
+INVALID_HRP = ["",
+               "bsadbhabdshabdsabsnmdsadmsabjfvasdmfmndsbfnmasdcdsnbcnmbcsdbcndsbnnbsnbdsnbnsbcnsdsn",
+               "   abcdefdhdhdh",
+               ]
+
+VALID_HRP = ["test",
+             "122333",
+             "?",
+             "\/",
+             "/!@#$%^?/><~"
+             ]
+
+
+class Testenc_dec(unittest.TestCase):
+
+    def test_encode_pure_bech32m(self):
```

Comment:
Split this into multiple functions to improve readability, I find it very hard to understand what you are testing.